Librarian: error trying to publish release

```

panic: runtime error: invalid memory address or nil pointer dereference
--
  | [signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xc7198e]
  |  
  | goroutine 1 [running]:
  | github.com/googleapis/librarian/internal/github.newClientWithHTTP({0xc00003e017, 0x5d}, 0x0, 0x0?)
  | /src/internal/github/github.go:62 +0x2e
  | github.com/googleapis/librarian/internal/github.NewClient(...)
  | /src/internal/github/github.go:57
  | github.com/googleapis/librarian/internal/automation.RunCommand({0x1109f38, 0x18b47e0}, {0x7ffda5cefe4d, 0xf}, {0x7ffda5cefe2a, 0x18}, 0x1, 0x1, 0x0)
  | /src/internal/automation/trigger.go:77 +0x16e
  | github.com/googleapis/librarian/internal/automation.Run({0xc00011a130?, 0xc000106068?, 0xc00005c740?})
  | /src/internal/automation/cli.go:35 +0x137
  | main.main()
  | /src/cmd/automation/main.go:29 +0x49


```

--- a/internal/github/github.go
+++ b/internal/github/github.go
@@ -59,7 +59,7 @@ func NewClient(accessToken string, repo *Repository) *Client {
 
 func newClientWithHTTP(accessToken string, repo *Repository, httpClient *http.Client) *Client {
 	client := github.NewClient(httpClient)
-	if repo.BaseURL != "" {
+	if repo != nil && repo.BaseURL != "" {
 		baseURL, _ := url.Parse(repo.BaseURL)
 		// Ensure the endpoint URL has a trailing slash.
 		if !strings.HasSuffix(baseURL.Path, "/") {

--- a/internal/github/github_test.go
+++ b/internal/github/github_test.go
@@ -1118,3 +1118,42 @@ func TestCreateTag(t *testing.T) {
 		})
 	}
 }
+
+func TestNewClient(t *testing.T) {
+
+	t.Parallel()
+	for _, test := range []struct {
+		name  string
+		token string
+		repo  *Repository
+	}{
+		{
+			name:  "with credentials",
+			token: "some-token",
+		},
+		{
+			name:  "with custom repo",
+			token: "some-token",
+			repo: &Repository{
+				Owner:   "some-owner",
+				Name:    "some-repo",
+				BaseURL: "https://example.com/",
+			},
+		},
+		{
+			name:  "with repo, without base url",
+			token: "some-token",
+			repo: &Repository{
+				Owner: "some-owner",
+				Name:  "some-repo",
+			},
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			client := NewClient(test.token, nil)
+			if client == nil {
+				t.Fatalf("expected to create a new client")
+			}
+		})
+	}
+}

